### ISSUE TITLE
Inconsistent cop names `Lint/DisjunctiveAssignmentInConstructor` and `Lint/OrAssignmentToConstant`

### ISSUE THREAD

**Issue by dvandersluis** [rubocop/rubocop#10089]
Both of these cops relate to the `||=` operator, but have different names ("disjunctive assignment" vs "or assignment"). They should be made consistent.

**Comment by dvandersluis** [comment]
@bbatsov I'm not sure if this is something that can be done before 2.0?

**Comment by bbatsov** [comment]
Yep, this will have to wait unless we figure out a way to be able to have a primary cop name and some alias for the sake of not breaking user configuration when renaming stuff. I guess we should start some list of all the cops we'd like to rename for one reason or another.

**Comment by bbatsov** [comment]
We should also agree which name to adopt as the canonical one. I assume that disjunctive assignment is the "proper" term, but I'm not sure about this.

**Comment by dvandersluis** [comment]
Agreed. FWIW I've never seen "disjunctive assignment" elsewhere, but it sounds fancier. It might be confusing to end-users though (and it was raised when the cop was first created - https://github.com/rubocop/rubocop/pull/6608#issuecomment-450776354)

**Comment by bbatsov** [comment]
Fair point - logical or is known as "disjunction", but I've heard this term used only in academic contexts. Perhaps we should stick to simpler terms.

**Comment by dvandersluis** [comment]
I think it's fine to stick with disjunctive, it's just more academic, but that's not a bad thing :grin:
